core/arch/x86/decode.c

```diff
@@ -777,6 +777,8 @@ read_evex(byte *pc, decode_info_t *di, byte instr_byte,
         NULL_TERMINATE_BUFFER(pc_addr);
         DO_ONCE(SYSLOG(SYSLOG_ERROR, AVX_512_SUPPORT_INCOMPLETE, 2,
                        get_application_name(), get_application_pid(), pc_addr));
+        if (ZMM_ENABLED())
+            dynamo_preserve_zmm_caller_saved = true;
```

Comment:
Yes, that sounds good.

Reply:
Discussed offline. See update.

Reply:
> Adds internal function instr_may_write_zmm_register, which conservatively decides whether the instruction is evex-encoded or writes to a zmm register.
 
> Moves setting sticky flag dynamo_preserve_zmm_caller_saved out of decoder and fast decoder, into bb builder and mangling. In bb builder, we check the prefix that may be set by decode_cti or somebody else. We do not break the bundle in case it was decode_cti.
    
> In mangling, we use the new function instr_may_write_zmm_register in order to turn on preserve_zmm_caller_saved.